tests/unit_tests/simtel/test_simtel_histograms.py

```diff
@@ -21,17 +27,259 @@
+    assert triggered_events_histogram["content_inside"] == 1.0
+
+    # Test defect histograms
+    new_instance = copy.copy(simtel_array_histogram_instance)
+    for i_hist, hist in enumerate(new_instance.histogram):  # altering intentionally the ids
+        if hist["id"] == 1:
+            new_instance.histogram[i_hist]["id"] = 99
+        if hist["id"] == 2:
+            new_instance.histogram[i_hist]["id"] = 99
+    with caplog.at_level(logging.ERROR):
+        with pytest.raises(HistogramIdNotFound):
+            new_instance.fill_event_histogram_dicts()
+    assert "Histograms ids not found. Please check your files." in caplog.text
+
+
+def test_produce_triggered_to_sim_fraction_hist(simtel_array_histogram_instance):
+    events_histogram = {"data": np.array([[10, 20, 30], [40, 50, 60]])}
+    triggered_events_histogram = {"data": np.array([[5, 10, 15], [20, 25, 30]])}
+
+    result = simtel_array_histogram_instance._produce_triggered_to_sim_fraction_hist(
+        events_histogram, triggered_events_histogram
+    )
+    expected_result = np.array([0.5, 0.5])
+
+    assert result == pytest.approx(expected_result, 0.01)
+
+    # Additional assertions
+    assert np.all(result >= 0)  # Check if all values are non-negative
+    assert np.all(result <= 1)  # Check if all values are less than or equal to 1
+
+
+def test_initialize_histogram_axes(simtel_array_histogram_instance):
+    events_histogram = {
+        "lower_x": 0,
+        "upper_x": 10,
+        "n_bins_x": 5,
+        "lower_y": 1,
+        "upper_y": 3,
+        "n_bins_y": 4,
+    }
+    simtel_array_histogram_instance._initialize_histogram_axes(events_histogram)
+    expected_radius_axis = np.linspace(0, 10, 6)
+    expected_energy_axis = np.logspace(1, 3, 5)
+    assert np.array_equal(simtel_array_histogram_instance.radius_axis, expected_radius_axis)
+    assert np.array_equal(simtel_array_histogram_instance.energy_axis, expected_energy_axis)
+
+
+def test_get_particle_distribution_function(simtel_array_histogram_instance):
+    # Test reference distribution function
+    reference_function = simtel_array_histogram_instance.get_particle_distribution_function(
+        label="reference"
+    )
+    assert isinstance(
+        reference_function, PowerLaw
+    )  # Assuming PowerLaw is the expected type for the reference function
+
+    # Test simulation distribution function
+    simulation_function = simtel_array_histogram_instance.get_particle_distribution_function(
+        label="simulation"
+    )
+    assert simulation_function.index == -2.0
+
+    # Test invalid label
+    with pytest.raises(ValueError):
+        simtel_array_histogram_instance.get_particle_distribution_function(label="invalid_label")
+
+
+def test_integrate_in_energy_bin(simtel_array_histogram_instance):
+    result = simtel_array_histogram_instance._integrate_in_energy_bin(
+        IRFDOC_PROTON_SPECTRUM, np.array([1, 10])
+    )
+    assert pytest.approx(result.value, 0.1) == 5.9e-06
```

Comment:
Is this 'epsilon' for pytest.approx in the right order of magnitude? 0.1 vs a value of 5.9e-6?

Reply:
The second argument is always a percentage (`rel`), see e.g. [link](https://docs.pytest.org/en/latest/reference/reference.html#pytest.approx)

Reply:
This is clear, but do you really want to test `5.9e-6 +- 0.001` ? So anything between in the interval [-0.0009941, 0.0010059]`?

Reply:
Again, 0.001 is relative! 5.9e-6 + 5.9e-6 * 0.001 and 5.9e-6 - 5.9e-6 * 0.001

>By default, approx considers numbers within a relative tolerance of 1e-6 (i.e. one part in a million) of its expected value to be equal. 
[https://docs.pytest.org/en/latest/reference/reference.html](url)

Reply:
OK - I suggest for improved readability to use `rel=0.1`.